src/scenes/Login/index.tsx

```diff
@@ -1,13 +1,31 @@
 import React from 'react';
-import { sleep } from '../../util';
+import { LoginMutationResult, useLoginMutation } from '../../api';
 import { LoginForm } from './LoginForm';
 
 export const Login = () => {
+  const [login] = useLoginMutation();
+
   return (
     <LoginForm
       onSubmit={async (data) => {
-        await sleep(1000);
-        alert(JSON.stringify(data, undefined, 2));
+        try {
+          const res: LoginMutationResult['data'] = await login({
+            variables: {
+              email: data.email,
+              password: data.password,
+            },
+          });
+
+          // TODO: post-login authentication (session storage, etc)
+          if (res?.login.success) {
+            alert(`Welcome ${res.login.user?.realFirstName.value}`);
+          } else {
+            alert('Login failed. Please check your email or password.');
+          }
+        } catch (e) {
+          alert('Login failed. Please contact support.');
+          console.log(e);
+        }
```

Comment:
I know this is still placeholder. But errors can and should be returned as strings or `{ [fieldName]: string }` objects.

Reply:
Can you elaborate here? Not sure I understand, sorry!

Reply:
I'm not saying you need to change anything right now. Eventually we'll want replace the failures alerts with a return value. Which I was trying to say final-form will interrupt as errors to display.

Reply:
Ohh got it! Actually, this reminds me of another thing we need to keep in mind: error formatting in apollo

Reply:
Yeah I hoping that now that we are actually consuming the API we'll better know how to do errors from the server.